unity-client/Assets/Scripts/MainScripts/DCL/Components/Audio/DCLAudioStream.cs

```diff
@@ -147,5 +143,20 @@ public void UpdateOutOfBoundariesState(bool enable)
                 Interface.WebInterface.SendAudioStreamEvent(model.url, true, 0);
             }
         }
+
+        public override void SetModel(object model)
+        {
+            Model prevModel = this.model;
+            this.model = (Model)model;
+            ApplyCurrentModel(prevModel);
+        }
+
+        private void ApplyCurrentModel(Model prevModel)
+        {
+            bool forceUpdate = prevModel.volume != model.volume;
+            settingsVolume = Settings.i.generalSettings.sfxVolume;
+
+            UpdatePlayingState(forceUpdate);
+        }
```

Comment:
I find it a bit confusing that we feed the `prevModel` to the `ApplyCurrentModel()` and also having the parameter called `prevModel` inside the `ApplyCurrentModel()` implementation.
Shouldn't we just apply the current model as we do in the other classes `SetModel` ?

Reply:
I have changed it to be more clear! The original thought to make the call `SetModel `was to have a way to set the model from unity without the need to always wait for a frame in the `ApplyChanges`. This way we can work with updated data if we create the component from Unity and set its data. 

Since `ApplyChanges ` required a JSON It doesn't have much sense to encode the data in Unity to work with Unity.  I was just searching for an easy way to make it work in all components without a major refactor in the code that supports the json and data.

If this doesn't work for you or you have a better idea, feel free to share it with me so we can take a look!